lib/AST/Decl.cpp

```diff
@@ -3827,6 +3827,18 @@ bool ValueDecl::isUsableFromInline() const {
   return false;
 }
 
+bool ValueDecl::skipAccessCheckIfInterface(const DeclContext *useDC,
+                                           AccessLevel useAcl,
+                                           AccessScope declScope) const {
+  if (!useDC || useAcl != AccessLevel::Package || !declScope.isPackage() ||
+      !isUsableFromInline() ||
+      getDeclContext()->getParentModule()->getBaseIdentifier() ==
```

Comment:
This check determining whether the decl and the use are in the same module, right? I think I usually see that done by doing a pointer comparison of the `ModuleDecl *` values. Is there a reason the identifier strings need to be compared?